### PR TITLE
Only remove terrain if `this.map.transform` exists

### LINKED ISSUE #1254: Layout benchmark broken
The Layout benchmark seems to be broken. Thanks @ambientlight for pointing this out.

Steps to reproduce the problem:
```
npm ci
npm run generate-typings && npm run build-prod && npm run build-csp && npm run build-dev && npm run build-css && npm run build-style-spec

npm run start-bench
```
(maybe I should put all the build commands in a build-all?)

Then open your browser, got to http://localhost:9966/bench/versions and wait for the Layout benchmark to show up (~~side note, one should be able to open http://localhost:9966/bench/versions?compare=main#Layout to see only the layout benchmark, but this is broken, too~~ works on a clean install).  Open the browser console and you will find the following error message

```
style.ts:503 Uncaught TypeError: Cannot read properties of undefined (reading 'updateElevation')
    at oe.setTerrain (style.ts:503:32)
    at oe._load (style.ts:287:14)
    at style.ts:242:18
```

Going forward, we should really refactor the benchmarks and run them in CI. This would maybe not give significant numbers, but at least we could see if the benchmarks still work.

### PR DESCRIPTION
**Pull request by wipfli**
## Launch Checklist

Fixes #1254

I wonder if there are other places where we have to be more careful that `this.map.transform` actually exists.

And should I add a unit test for this?

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [ ] Manually test the debug page.
 - [ ] Suggest a changelog category: bug/feature/docs/etc. or "skip changelog".
 - [ ] Add an entry inside this element for inclusion in the `maplibre-gl-js` changelog: `<changelog></changelog>`.

### CODE REVIEW

**Comment by HarelM** [comment]
I think transform must exist always, shouldn't it?

**Comment by wipfli** [comment]
I would have thought so too. I added some debug print statement. Here are the steps to reproduce this bug:

1. clone this pull request
2. npm run ci
3. npm run start-bench
4. open web browser at http://localhost:9966/bench/versions?#Layout
5. console will show 

```
this.map is Cr {_requestManager: s, style: oe}
this.map.transform is undefined
```

**Comment by HarelM** [comment]
I see. The only question I have in that case is if there's something that is not fully defined in the benchmark code that causes this.
I don't believe this is something that happens in production code so I tend to say this is probably an orchestration code issue...?

**Comment by wipfli** [comment]
Probably you are right @HarelM. I am reading a bit through the benchmark code. Hard to follow what it does for me. I wonder if we should refactor it from `Promise` + `.then()` to `async` + `await` style...

**Comment by wipfli** [comment]
Turns out the problem was that the `MapStub` class used by the benchmarks did not have the transform property. So your guess was correct, @HarelM, thanks for the hint! Fixed in #1297
